pkg/controller/common.go

```diff
@@ -477,6 +495,34 @@ func cleanUpAzVolumeAttachments(ctx context.Context, azr azReconciler, attachmen
 	return nil
 }
 
+func getAzVolumeAttachmentsForVolume(ctx context.Context, azclient client.Client, volumeName, role string) (attachments []v1alpha1.AzVolumeAttachment, err error) {
```

Comment:
Consider using something similar to `cleanUpMode` for `role` parameter. This would help clarify function usage and help reduce unclarity.